src/Disks/ObjectStorages/DiskObjectStorageTransaction.cpp

```diff
@@ -1177,25 +1210,69 @@ TransactionCommitOutcomeVariant DiskObjectStorageTransaction::tryCommit(const Tr
         throw;
     }
 
+    /// after successful commit of metadata transaction we can not rollback transaction
+
     StoredObjects objects_to_remove;
     for (const auto & operation : operations_to_execute)
-        operation->finalize(objects_to_remove);
+    {
+        try
+        {
+            operation->finalize(objects_to_remove);
```

Comment:
this in general can't throw anything

Reply:
but it is not declared as `nothrow`, then it is better to wrap with try catch then.